lib/common/tls.c

```diff
@@ -384,6 +384,14 @@ pcmk__read_handshake_data(const pcmk__client_t *client)
          * invoked again once the client sends more.
          */
         return EAGAIN;
+    } else if (rc == GNUTLS_E_CERTIFICATE_VERIFICATION_ERROR) {
+        int type = gnutls_certificate_type_get(client->remote->tls_session);
```

Comment:
I'd prefer to see this as a static function, but I'm not going to hold up the PR over that. If you feel like it, add it to a later PR.